### PR TITLE
Add V2 product image

### PR DESCRIPTION
**Pull request by szeber**

### CODE REVIEW

File: src/ShoppinPal/Vend/DataObject/Entity/EntityDoAbstract.php [hunk 1]
```diff
@@ -217,4 +181,56 @@ protected function getProperties()
...
+     * @param string $unknownPropertyHandling
+     *
+     * @return void
+     * @throws ParameterException
+     */
+    protected function setProperty($key, $value, $unknownPropertyHandling = self::UNKNOWN_PROPERTY_THROW)
+    {
+        $propertyName = StringHelper::underScoreToCamel($key);
+
+        if (!property_exists($this, $propertyName)) {
```
Comment (emulgeator) [inline src/ShoppinPal/Vend/DataObject/Entity/EntityDoAbstract.php:197]: This whole block can go to a separate method

File: src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImage.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ];
+
+    protected $subEntities = [
+        'sizes' => [
+            self::SUB_ENTITY_KEY_TYPE  => self::SUB_ENTITY_TYPE_SINGLE,
+            self::SUB_ENTITY_KEY_CLASS => ProductImageSize::class,
+        ],
+    ];
+
+    public $id;
```
Comment (emulgeator) [inline src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImage.php]: Document the type of the properties

File: src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImageSize.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        'sl',
+        'sm',
+        'ss',
+        'st',
+        'thumb',
+        'original',
+        'standard',
+    ];
+
+    public $sl;
```
Comment (emulgeator) [inline src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImageSize.php]: Types missing.
Also I know these are coming from Vend but more descriptive names would be great or at least some docs about what do these mean exactly

**Review by emulgeator — APPROVED** [review]
Small changes needed but LGTM

File: src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImageSize.php (hunk 3 above)
Comment (szeber) [reply src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImageSize.php]: Problem is, I haven't got the slightest clue about that :D

File: src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImage.php (hunk 2 above)
Comment (szeber) [reply src/ShoppinPal/Vend/DataObject/Entity/V2/ProductImage.php]: Done it for the new entities, but we'll have to document the other entities as well later. Also we should add more docs on the properties that are not exactly clear.

**Pull request merged**